### ISSUE TITLE
Video not fullscreen

### ISSUE THREAD

**Issue by tricki** [JohnCoates/Aerial#1350]
#### General troubleshooting tips

Before logging an issue please check that:
- [x] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [x] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [x] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [x] Mac model: MacBook Pro, 16-inch, 2021
- [x] macOS version: 14.1.1 (23B81)
- [x] Monitor setup: Primary: Dell U220Q 3840 × 2160, Secondary: Built-in Liquid Retina XDR Display 3456 × 2234

- [x] If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue

Sometimes the video is not fullscreen on my primary (external) monitor, but stuck to the bottom left corner at roughly half the screen's width (see screenshot). It always happens when I return after a break, so I'm not sure whether it starts like that or whether it starts correctly and then breaks. This doesn't happen on the secondary (built-in) screen.

I've had this issue for a while, just never took the time to report it. I've also had some other issues after updating to Sonoma (e.g. screen was always black) but those seem to have resolved themselves.

I activated Debug Mode and manually launched Aerial (using the Companion's "Launch Now" option). It opened correctly 5 times, but the 6th time I could reproduce the bug. One thing that sticks out from the log is 42 instances of `🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨 The data couldn’t be read because it isn’t in the correct format.`, but I'm not sure what data this is referring to.

Here are my "Display" settings:

<img width="804" alt="image" src="https://github.com/JohnCoates/Aerial/assets/523154/e6a0655f-f50d-416a-a118-e333d1dbf764">

Screenshot:

![Screenshot 2023-11-30 at 10 47 36](https://github.com/JohnCoates/Aerial/assets/523154/18b74068-9e66-4475-ac2d-fb2b79d545de)

**Comment by tricki** [comment]
By bad for not including the log, I had it prepared and everything 🤦

I had deleted the existing log and manually launched Aerial 6 times until the bug showed up, then immediately disabled the logging.

[AerialLog.txt](https://github.com/JohnCoates/Aerial/files/13526685/AerialLog.txt)

<img width="610" alt="image" src="https://github.com/JohnCoates/Aerial/assets/523154/cc58fbd9-7f21-4081-a77f-b64f36015557">

No worries, I've had my share of problems with Apple (just App Store related). 

If you need more info, I'm happy to help.

**Comment by glouel** [comment]
No worries thanks I'll have a look. Your original arrangement should be ok so please go back to it. The 2nd one is the one causing some issues at the moment. I'll get back to you asap !

**Comment by tricki** [comment]
Ah yes, I have a custom scaling, completely forgot about that.
I'm not sure where I can get a numberical factor, but here's a screenshot of the external monitor:

![image](https://github.com/JohnCoates/Aerial/assets/523154/4f19ad98-c5a9-41d2-b97c-8d0e9d9a1356)

<img width="463" alt="image" src="https://github.com/JohnCoates/Aerial/assets/523154/6473cc55-f509-483b-83ed-c97b5a6c8172">

I've done some tests with the other scaling options and was able to reproduce the problem on all but "Larger Text" (1920x1080). But depending on which one is selected, the portion of the screen that is black is different.

FYI I didn't see your "Edit:" until today, since I didn't receive a second notification.

**Comment by glouel** [comment]
Yeah, my bad, I tend to edit when I just posted it but people who rely on emails do miss them. 

I have to say I'm a bit puzzled, because you mention that only 1920x1080 works as expected, which is itself a scaling. When scaling messes up, usually only the middle option (which is native resolution) is the one that works ! 

It also looks like I'm correctly reading your resolution (3360.0 x 1890.0) and there's no particular issue or .8something pixels, so maybe I'm thinking about all this wrong ? 

I'll try to play around with my monitors here, see if I can trigger anything, but I had another idea in the meantime, can you remove the 2cm margin bottom right here ? 

![Capture d’écran 2023-12-04 à 16 55 11](https://github.com/JohnCoates/Aerial/assets/37544189/0f46fa59-eca3-4987-a8d8-7beee8002754)

It's possible that under Sonoma the many workarounds broke that feature, maybe. Thanks !

**Comment by glouel** [comment]
Ok I've tried to dig this more this morning, I tried practically every resolution mismatch between my 2 monitors and can't seem to reproduce anything. I also tried many many variations of margin alignments and it's always correct.

I ended up digging more in your logs and I think there might be a clue around here. You mentioned your last run was bugged and the others weren't. I did first a cursory check trying to match each instance of AerialViews, from their init to their startAnimation, see if somehow stuff got corrupted . 

```
2023-11-30 10:58:19.441 : 🖼️ AVinit (.saver) (0.0, 0.0, 2056.0, 1329.0) p: false o: false
2023-11-30 10:58:19.442 : 🖼️ <AerialView: 0x12821afa0> viewDidMoveToWindow frame: (0.0, 0.0, 2056.0, 1329.0) window: Optional(<NSServiceViewControllerWindow: 0x128110a70>)
...
2023-11-30 10:58:19.963 : 🖼️ <AerialView: 0x12821afa0> startAnimation frame (0.0, 0.0, 2056.0, 1329.0) bounds (0.0, 0.0, 2056.0, 1329.0)
```

That's your internal screen

```
2023-11-30 10:58:19.456 : 🖼️ AVinit (.saver) (0.0, 0.0, 3360.0, 1890.0) p: false o: false
2023-11-30 10:58:19.457 : 🖼️ <AerialView: 0x127743250> viewDidMoveToWindow frame: (0.0, 0.0, 3360.0, 1890.0) window: Optional(<NSServiceViewControllerWindow: 0x128629390>)
2023-11-30 10:58:19.963 : 🖼️ <AerialView: 0x127743250> startAnimation frame (0.0, 0.0, 3360.0, 1890.0) bounds (0.0, 0.0, 3360.0, 1890.0)
```

That's your external Dell

So at the very least, Aerial is internally consistent, we get told by macOS we are on X by Y screen, and we play such a resolution. 

I do get a weird discrepancy though on your failed run, and I'm not quite sure what to make of it : 

```
2023-11-30 10:58:19.456 : 🖼️ AVinit (.saver) (0.0, 0.0, 3360.0, 1890.0) p: false o: false
2023-11-30 10:58:19.457 : 🖼️ <AerialView: 0x127743250> viewDidMoveToWindow frame: (0.0, 0.0, 3360.0, 1890.0) window: Optional(<NSServiceViewControllerWindow: 0x128629390>)
2023-11-30 10:58:19.457 : Optional(<NSScreen: 0x12843e980; name="Built-in Retina Display"; backingScaleFactor=2.000000; frame={{-2056, 141}, {2056, 1329}}; visibleFrame={{-2056, 141}, {2056, 1285}}>)
```

- So when we get initialized, macOS tells us we are on your Dell. 
- We get "moved" to a window, and our frame still says Dell
- I do log the monitor name for extra debugging  and... we're told we are on your built-in, but with the size of your Dell

So what happens to you is, when the bug happens, only the bottom right portion of the video plays on your internal), and on the dell the video gets played at the size of your internal display, on the big one.
I would also expect that the left part of the video plays on the right screen. 

So basically, macOS mixes up screens (again, I've been fighting this for 10 months now ?) but in a new fancy way I hadn't seen before. I do have an idea for a tentative fix though. 

Would you be willing to try a test build if I get my fix running ? Thanks !

**Comment by tricki** [comment]
It's a company Mac, so I'd have to get permission for that.

But interestingly, after an OS update to 14.1.2 I cannot reproduce the problem anymore. Maybe the update fixed a bug, or maybe my system was corrupt in some way that the update reset...

Anyway, thanks so much for the help and effort! I'll close this for now, but will reopen if it happens again.

If you'd like some more info (e.g. another log) I'm happy to help.

**Comment by glouel** [comment]
That's interesting, thanks for the update. Let me know if it somehow reappears.
